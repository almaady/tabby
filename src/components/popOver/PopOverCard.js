import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import ButtonIcon from "../buttons/ButtonIcon";
import PopOverHeader from "./PopOverHeader";
/**
 * @class PopOver Card - Is the card for the pop over and it contains a cross icon by default
 */
class PopOverCard extends React.PureComponent {
  /**
   * @property {bool} show - Determines if the pop over is visible
   * @property {func} onClose - The function to close the modal
   * @property {node} children - The content of the modal
   * @property {string} className - Just in case you need another class
   * @property {string} title - The title of the pop over
   * @property {bool} back - Determines if the pop over has a back button
   * @property {func} onBack - The function to go back
   * @property {number} left - The value of the left position of the pop over card
   * @property {number} top - The value of the top position of the pop over card
   */
  static propTypes = {
    show: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    title: PropTypes.string,
    onBack: PropTypes.func,
    back: PropTypes.bool,
    className: PropTypes.string,
    left: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    top: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
  };

  render() {
    const {
      onClose,
      children,
      className,
      show,
      title,
      onBack,
      back,
      left,
      top
    } = this.props;
    return (
      <div
        style={{
          left,
          top
        }}
        className={classNames("pop-over", className, {
          "visibility-hidden": !show
        })}
      >
        <div className="icons">
          <ButtonIcon
            icon="left"
            onClick={onBack}
            className={classNames({ "visibility-hidden": !back })}
            size="R"
          />
          <PopOverHeader title={title} />
          <ButtonIcon 
            icon="cross" 
            size="R"
            onClick={onClose} />
        </div>
        <div className="pop-over--content">{children}</div>
      </div>
    );
  }
}

PopOverCard.defaultProps = {
  className: "",
  title: "",
  back: false,
  onBack: null,
  left: "",
  top: ""
};

export default PopOverCard;
