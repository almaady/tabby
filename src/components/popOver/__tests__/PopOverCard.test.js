import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import PopOverCard from "../PopOverCard";

describe("PopOver", () => {
  const close = sinon.stub();
  const back = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      show: false,
      children: <div  />,
      onClose: close,
      onBack: back
    };
  });

  it("should render correctly with default state", () => {
    const component = shallow(<PopOverCard {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should add visibility-hidden class when show is false", () => {
    const compomnent = shallow(<PopOverCard {...props} />);
    expect(compomnent.hasClass("visibility-hidden")).toEqual(true);
  });
  it("should call OnClose when you click on button", () => {
    const component = shallow(<PopOverCard {...props} />);
    component
      .find("ButtonIcon")
      .last()
      .simulate("click");
    expect(close.called).toBeTruthy();
  });
  it("should call OnBack when you click on button", () => {
    const component = shallow(<PopOverCard {...props} />);
    component
      .find("ButtonIcon")
      .first()
      .simulate("click");
    expect(back.called).toBeTruthy();
  });
  it("should call onClose when you click ", () => {
    const component = shallow(<PopOverCard {...props} />);
    !component.simulate("click");
    expect(close.called).toBeTruthy();
  });
});
