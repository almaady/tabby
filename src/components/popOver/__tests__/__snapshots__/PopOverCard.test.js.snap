// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopOver should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopOverCard
    back={false}
    className=""
    left=""
    onBack={[Function]}
    onClose={[Function]}
    show={false}
    title=""
    top=""
  >
    <div />
  </PopOverCard>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icons"
        >
          <ButtonIcon
            borderColor=""
            className="visibility-hidden"
            color="gray-l1"
            disabled={false}
            icon="left"
            onClick={[Function]}
            size="R"
            type="button"
          />
          <PopOverHeader
            title=""
          />
          <ButtonIcon
            borderColor=""
            className=""
            color="gray-l1"
            disabled={false}
            icon="cross"
            onClick={[Function]}
            size="R"
            type="button"
          />
        </div>,
        <div
          className="pop-over--content"
        >
          <div />
        </div>,
      ],
      "className": "pop-over visibility-hidden",
      "style": Object {
        "left": "",
        "top": "",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ButtonIcon
              borderColor=""
              className="visibility-hidden"
              color="gray-l1"
              disabled={false}
              icon="left"
              onClick={[Function]}
              size="R"
              type="button"
            />,
            <PopOverHeader
              title=""
            />,
            <ButtonIcon
              borderColor=""
              className=""
              color="gray-l1"
              disabled={false}
              icon="cross"
              onClick={[Function]}
              size="R"
              type="button"
            />,
          ],
          "className": "icons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "borderColor": "",
              "className": "visibility-hidden",
              "color": "gray-l1",
              "disabled": false,
              "icon": "left",
              "onClick": [Function],
              "size": "R",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "borderColor": "",
              "className": "",
              "color": "gray-l1",
              "disabled": false,
              "icon": "cross",
              "onClick": [Function],
              "size": "R",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "pop-over--content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icons"
          >
            <ButtonIcon
              borderColor=""
              className="visibility-hidden"
              color="gray-l1"
              disabled={false}
              icon="left"
              onClick={[Function]}
              size="R"
              type="button"
            />
            <PopOverHeader
              title=""
            />
            <ButtonIcon
              borderColor=""
              className=""
              color="gray-l1"
              disabled={false}
              icon="cross"
              onClick={[Function]}
              size="R"
              type="button"
            />
          </div>,
          <div
            className="pop-over--content"
          >
            <div />
          </div>,
        ],
        "className": "pop-over visibility-hidden",
        "style": Object {
          "left": "",
          "top": "",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ButtonIcon
                borderColor=""
                className="visibility-hidden"
                color="gray-l1"
                disabled={false}
                icon="left"
                onClick={[Function]}
                size="R"
                type="button"
              />,
              <PopOverHeader
                title=""
              />,
              <ButtonIcon
                borderColor=""
                className=""
                color="gray-l1"
                disabled={false}
                icon="cross"
                onClick={[Function]}
                size="R"
                type="button"
              />,
            ],
            "className": "icons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "borderColor": "",
                "className": "visibility-hidden",
                "color": "gray-l1",
                "disabled": false,
                "icon": "left",
                "onClick": [Function],
                "size": "R",
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "borderColor": "",
                "className": "",
                "color": "gray-l1",
                "disabled": false,
                "icon": "cross",
                "onClick": [Function],
                "size": "R",
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "pop-over--content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
